fedimint-core/src/config.rs

```diff
@@ -120,6 +120,8 @@ pub struct PeerUrl {
 /// This includes global settings and client-side module configs.
 #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Encodable, Decodable)]
 pub struct ClientConfig {
+    #[serde(default)]
+    pub federation_id: Option<FederationId>,
```

Comment:
Why `Option`? To make it backwards compatible?

Reply:
Yes, happy to change and add versioning but this seemed more straightforward

Reply:
Add a `// support(v0.2): can be changed into non-option` so we don't forget